config/server.go

```diff
@@ -34,19 +35,56 @@ type serverGRPC struct {
 	Port string
 }
 
+type serverCors struct {
+	Enabled          bool
+	AllowedOrigin    []string
+	AllowedMethods   []string
+	AllowedHeaders   []string
+	AllowCredentials bool
+	MaxAge           int
+}
+
 func serverConfigLoader() {
 	viper.SetDefault("SERVER_BATCH_DEDUP_IN_CONNECTION_ENABLED", "false")
 	Server = server{
 		DedupEnabled: util.MustGetBool("SERVER_BATCH_DEDUP_IN_CONNECTION_ENABLED"),
 	}
 }
 
+func serverCorsConfigLoader() {
+	allowedHeaders := []string{"Content-Type"}
+	allowedHeaders = setAllowedHeaders(allowedHeaders, "SERVER_WEBSOCKET_CONN_GROUP_HEADER", "")
+	allowedHeaders = setAllowedHeaders(allowedHeaders, "SERVER_WEBSOCKET_CONN_ID_HEADER", "")
+	viper.SetDefault("SERVER_CORS_ENABLED", false)
+	viper.SetDefault("SERVER_CORS_ALLOWED_ORIGIN", "*")
+	viper.SetDefault("SERVER_CORS_ALLOWED_METHODS", []string{"GET", "HEAD", "POST"})
+	viper.SetDefault("SERVER_CORS_ALLOWED_HEADERS", allowedHeaders)
+	viper.SetDefault("SERVER_CORS_ALLOW_CREDENTIALS", false)
+	viper.SetDefault("SERVER_CORS_PREFLIGHT_MAX_AGE_SECONDS", 0)
+	ServerCors = serverCors{
+		Enabled:          util.MustGetBool("SERVER_CORS_ENABLED"),
+		AllowedOrigin:    viper.GetStringSlice("SERVER_CORS_ALLOWED_ORIGIN"),
+		AllowedMethods:   viper.GetStringSlice("SERVER_CORS_ALLOWED_METHODS"),
+		AllowCredentials: util.MustGetBool("SERVER_CORS_ALLOW_CREDENTIALS"),
+		AllowedHeaders:   viper.GetStringSlice("SERVER_CORS_ALLOWED_HEADERS"),
+		MaxAge:           util.MustGetInt("SERVER_CORS_PREFLIGHT_MAX_AGE_SECONDS"),
+	}
+}
+
+func setAllowedHeaders(allowedHeaders []string, envKey string, defaultValue string) []string {
+	viper.SetDefault(envKey, defaultValue)
```

Comment:
This is a helper function to help populate headers used by raccoon for connection and user identification by default, instead of asking user to duplicate it.
We are setting defaults so that we can easily compare it for empty value.
If we do not setDefault we would would need to check for both nil and empty string as possible values to skip.

I guess checking nil and empty string would be better as setting default here can lead to unintended side effects when the default values needs an update

Reply:
Removed redundant default value set.